### PR TITLE
ServiceNow operations

### LINKED ISSUE #494: ServiceNow
Operations to consider:
- New record
- New/updated record
- New record (real-time)
- New/updated record (real-time)
- Custom action
- Create record
- Search record
- Update record
- Create record using a template
- Update record using a template

Specs: https://docs.google.com/presentation/d/10-WlU7QNErWaJ9QAd04KWUvWWsPnpXRLFsCUqqWjOpw/edit

### PR DESCRIPTION
**Pull request by eeshansim**
fixes #494

### CODE REVIEW

File: connectors/servicenow/create.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+      </td>
+    </tr>
+  </tbody>
+</table>
+
+### Output fields
+The output of this action is the full set of columns from the selected table. All default and custom columns are supported. The output fields will be displayed only after a table is provided, either by selecting a table from the pick list or by providing the full table name.
+
+## Create record using a template
+This action creates a single record into any table in your ServiceNow instance by applying a template to the newly created record. Using a template simplifies the process by populating fields automatically.
```
Comment (jandonyada) [inline connectors/servicenow/create.md]: We should mention that the templates values will be used to create the record only at runtime. Not sure if 'populating' is the word to use here as users may expect the fields to populate in the recipe itself, which is not the case. As discussed:

After a template is chosen, you can add still map datapills into the fields. The template values will be applied on top of your mapping. If you want to override the template values with your own, choose 'Yes' under 'Override template'.

File: connectors/servicenow/search.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  </thead>
+  <tbody>
+    <tr>
+      <td><a href="#table">Table</a></td>
+      <td>
+        Select a table to search records from.
+      </td>
+    </tr>
+    <tr>
+      <td><a href="#columns-to-search-by">Columns to search by</a></td>
```
Comment (jandonyada) [inline connectors/servicenow/search.md]: As discussed: 'Search by column values' sounds more natural.

File: connectors/servicenow/search.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+      <td>
+        Provide data to match records. Only records that match all the search criteria will be returned in the output.
+      </td>
+    </tr>
+  </tbody>
+</table>
+
+### Output fields
+The output of this action is a list of records that match your search criteria. The output datatree contains the full set of columns from the selected table. All default and custom columns are supported. The output fields will be displayed only after a table is provided, either by selecting a table from the pick list or by providing the full table name.
```
Comment (jandonyada) [inline connectors/servicenow/search.md]: Screenshot here would be good.

Also, the phrasing here is much better:
>The **output datatree** contains the full set of columns from the selected table.

Consider using this in the **Output fields** of the other documents.

File: connectors/servicenow/search.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+## Input field definitions
+
+### Table
+Select the table to search records from. This can be done either by selecting a table from the pick list, or toggling the input field to text mode and typing the full table name.
+
+Make sure that the connected user has [sufficient access control](/connectors/servicenow.md#roles-and-permissions-required-to-connect) to the selected table.
+
+### Columns to search by
```
Comment (jandonyada) [inline connectors/servicenow/search.md]: Search by column values

File: connectors/servicenow/update.md [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+      <td>
+        Select a table to update a record in.
+      </td>
+    </tr>
+    <tr>
+      <td><a href="#sys-id">Sys ID</a></td>
+      <td>Sys ID of the record to be updated.</td>
+    </tr>
+    <tr>
+      <td><a href="#columns-to-be-updated">Columns to be updated</a></td>
```
Comment (jandonyada) [inline connectors/servicenow/update.md]: Standardize this term across all docs once you've decided on which to use.

File: connectors/servicenow/update.md [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+      <td>
+        Provide data for each column of the record to be updated.
+      </td>
+    </tr>
+  </tbody>
+</table>
+
+### Output fields
+The output of this action is the full set of columns from the selected table. All default and custom columns are supported. The output fields will be displayed only after a table is provided, either by selecting a table from the pick list or by providing the full table name.
```
Comment (jandonyada) [inline connectors/servicenow/update.md]: As mentioned in search.md, use "The output datatree of this action..." - much clearer.

File: connectors/servicenow/update.md [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+
+## Update record using a template
+This action updates a single record into any table in your ServiceNow instance by applying a template to the record. Using a template simplifies the process by populating fields automatically.
```
Comment (jandonyada) [inline connectors/servicenow/update.md]: Standardize the paragraph here with whatever you decide in the previous docs.

For reference:
>We should mention that the templates values will be used to updatethe record only at runtime. Not sure if 'populating' is the word to use here as users may expect the fields to populate in the recipe itself, which is not the case. As discussed:

>After a template is chosen, you can add still map datapills into the fields. The template values will be applied on top of your mapping. If you want to override the template values with your own, choose 'Yes' under 'Override template'.

File: connectors/servicenow/update.md [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+      <td>
+        Choose whether you want to override the default template values with the data provided in <b>Columns to be updated</b>.
+      </td>
+    </tr>
+    <tr>
+      <td><a href="#sys-id">Sys ID</a></td>
+      <td>Sys ID of the record to be updated.</td>
+    </tr>
+    <tr>
+      <td><a href="#columns-to-be-updated">Columns to be updated</a></td>
```
Comment (jandonyada) [inline connectors/servicenow/update.md]: Standardize term

File: connectors/servicenow/update.md [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+      <td>
+        Provide data for each column of the record to be updated.
+      </td>
+    </tr>
+  </tbody>
+</table>
+
+### Output fields
+The output of this action is the full set of columns from the selected table. All default and custom columns are supported. The output fields will be displayed only after a table is provided, either by selecting a table from the pick list or by providing the full table name.
```
Comment (jandonyada) [inline connectors/servicenow/update.md]: The output datatree of this action

File: connectors/servicenow/updated-record-trigger.md [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+      <td><a href="#table">Table</a></td>
+      <td>
+        Select a table to pick up new or updated records from.
+      </td>
+    </tr>
+  </tbody>
+</table>
+
+### Output fields
+The output of this trigger is a full set of columns from the selected table. All default and custom columns are supported. The output fields will be displayed only after a table is provided, either by selecting a table from the pick list or by providing the full table name.
```
Comment (jandonyada) [inline connectors/servicenow/updated-record-trigger.md]: The output datatree of this trigger...

Comment (jandonyada) [reply connectors/servicenow/updated-record-trigger.md]: Screenshot

File: connectors/servicenow/updated-record-trigger.md [hunk 11]
```diff
@@ -0,0 +1,88 @@
...
+
+## New/updated record (real-time)
+This trigger picks up records in the selected table as soon they are created or updated. Each record is processed as a separate job. It makes use of business rules to detect and trigger a new or updated record event to be processed.
+
+**To use this trigger, the connected user must have sufficient access control to sys_script table. Learn how to configure it [here](/connectors/servicenow.md#real-time-trigger).**
```
Comment (jandonyada) [inline connectors/servicenow/updated-record-trigger.md]: Consider naming the link in full e.g.

Learn how to configure it in our [ServiceNow documentation](/connectors/servicenow.md#real-time-trigger).

or something like that =)

File: connectors/servicenow/updated-record-trigger.md [hunk 12]
```diff
@@ -0,0 +1,88 @@
...
+      <td><a href="#table">Table</a></td>
+      <td>
+        Select a table to pick up new or updated records from.
+      </td>
+    </tr>
+  </tbody>
+</table>
+
+### Output fields
+The output of this trigger is a full set of columns from the selected table. All default and custom columns are supported. The output fields will be displayed only after a table is provided, either by selecting a table from the pick list or by providing the full table name.
```
Comment (jandonyada) [inline connectors/servicenow/updated-record-trigger.md]: The output datatree of this trigger...

Comment (jandonyada) [reply connectors/servicenow/updated-record-trigger.md]: Screenshot

File: connectors/servicenow/updated-record-trigger.md [hunk 13]
```diff
@@ -0,0 +1,88 @@
...
+
+## Input field definitions
+
+### From
+This is a user-defined time that marks the earliest time that the trigger should pick up created or updated records. If left blank, the default time will be set to <b>one day</b> before the recipe is first started.
+
+The **sys_updated_on** column is used to determine the time that a record is created or updated. Only records with **sys_updated_on** timestamp later than the time specified in **From** will be processed by the trigger.
```
Comment (jandonyada) [inline connectors/servicenow/updated-record-trigger.md]: Screenshot would be good - not sure where to look for **sys_updated_on**

File: connectors/servicenow/new-record-trigger.md [hunk 14]
```diff
@@ -0,0 +1,88 @@
...
+  </tbody>
+</table>
+
+### Output fields
+The output of this trigger is a full set of columns from the selected table. All default and custom columns are supported. The output fields will be displayed only after a table is provided, either by selecting a table from the pick list or by providing the full table name.
+
+## New record (real-time)
+This trigger picks up records in the selected table as soon they are created. Each record is processed as a separate job. It makes use of business rules to detect and trigger a new record event to be processed.
+
+**To use this trigger, the connected user must have sufficient access control to sys_script table. Learn how to configure it [here](/connectors/servicenow.md#real-time-trigger).**
```
[the remark this answers is not in the document]
Comment (eeshansim) [reply connectors/servicenow/new-record-trigger.md]: Agree. Will change

**Pull request merged**